Sources/Shared/Algorithms/Heckel.swift

```diff
@@ -103,12 +100,12 @@ public final class Heckel: DiffAware {
       newArray.append(.tableEntry(entry))
 
       //
-      table[item.hashValue] = entry
+      table[item.idProviding] = entry
     }
   }
 
-  private func perform2ndPass<T: Hashable>(
-    old: Array<T>,
+  private func perform2ndPass(
+    old: [T],
     table: inout [Int: TableEntry],
```

Comment:
I'm curious, is there a reason why the key of table is Int rather than just any Hashable type? If it was any Hashable type then idProviding could also return a Hashable object to use for the key. That way you could use the id object itself, helping reduce any potential problems with collisions (after all, the value of hashValue is not guaranteed to be unique)

Reply:
@pilky yeah that is more intuitive, but `Hashable` can only be used as generic constraint, and here I need just `hashValue` which is `Int`, so I choose `Int` for simplicity for now